exporter/splunkhecexporter/tracedata_to_splunk.go

```diff
@@ -17,43 +17,149 @@ package splunkhecexporter
 import (
 	"go.opentelemetry.io/collector/consumer/pdata"
 	"go.opentelemetry.io/collector/translator/conventions"
-	"go.opentelemetry.io/collector/translator/internaldata"
 	"go.uber.org/zap"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/splunk"
 )
 
+// HecEvent is a data structure holding a span event to export explicitly to Splunk HEC.
+type HecEvent struct {
+	Attributes map[string]interface{}  `json:"attributes,omitempty"`
+	Name       string                  `json:"name"`
+	Timestamp  pdata.TimestampUnixNano `json:"timestamp"`
+}
+
+// HecLink is a data structure holding a span link to export explicitly to Splunk HEC.
+type HecLink struct {
+	Attributes map[string]interface{} `json:"attributes,omitempty"`
+	TraceID    []byte                 `json:"trace_id"`
+	SpanID     []byte                 `json:"span_id"`
```

Comment:
I think you need either base64 or base16 (hex) encoded correct? Not sure what is the right thing that you want.

Reply:
So probably worth making this string and call `HexString` on the ids?

Reply:
yeah I thought base64 was still the way. Apparently hex is now standard, moving to that.